Automattic/wp-calypso
# Gutenboarding: Add `/new/plans` route that shows plans grid

#### Changes proposed in this Pull Request

Adds plans step at `/new/plans`, removing the need for the `prelaunch` frankenflow.

Combined with #42026, this means users who choose the eCommerce plan will have their site transfered to atomic and will go through the store onboarding.

* Plan Step comes after font step if:
  * the user chooses a paid domain
  <s>* and the haven't already choosen a domain
    * either by using the Plans Modal button in the top right
    * or by specifying one as a route param</s> https://github.com/Automattic/wp-calypso/pull/42028#pullrequestreview-408976078
* Switching between plans immediately updates the store (can be seen by the plan label in the top-right).
  * This is <s>different from</s> now also the Plans Modal behaviour https://github.com/Automattic/wp-calypso/pull/42028#pullrequestreview-408976078
* User can go back to the Font Selection step
* The plans grid auto-selects the plan specified in the route
* The Plans Step fires tracks events
  * `calypso_newsite_step_enter`
  * `calypso_newsite_step_leave`, which has a `selected_plan` property
* Remove link to `prelaunch` flow

#### Testing instructions



* Testing requires the `gutenboarding/plans-grid` feature flag, so should be tested locally.
  * `/new`
  * Choose a paid domain
  * The next step after choosing a font should be the Plans Step
  * If there was no previous interaction with Plans Button, _Premium Plan_ should be pre-selected
  * Site should be created as expected:
    * If free plan then go straight to block editor
    * If paid plan then go to checkout, then block editor when complete
    * If eCommerce plan then go to checkout, then on to the store onboarding
* Also test on `calypso.live` to check gutenboarding isn't broken when the feature flag is disabled.
